### PR TITLE
Move compactView option to AccountStorageObject

### PR DESCRIPTION
**Pull request by kapraran**
This PR can act as an opportunity for discussion about how account object storage should be used.

I have some questions

1. How the accountStorage state is hydrated? Are you doing it from VU side?
2. Should we always update the accountStorage through the SET_ACCOUNT_STORAGE_VALUE or we are free to use the web call in another reducers?
3. Should the values in accountStorage all be strings (i use boolean now) and manually parse them when its time to be used in components?

### CODE REVIEW

**Comment by OrfeasZ** [comment]
1. Yes. Storage will be updated optimistically client-side and then VU will then update it with the final state after server-side confirmation with the `SET_ACCOUNT_STORAGE` action.
2. Yes. Use the actions.
3. Yes. Only strings are supported.

**Comment by kapraran** [comment]
Thanks. I'll do the type updates later.

File: app/reducers/user.js [hunk 1]
```diff
@@ -31,7 +32,9 @@ const initialState = {
     playerLoginStatus: 0,
     originLinkStatus: 0,
     loginToken: null,
-    accountStorage: {},
+    accountStorage: {
+        [AccountStorageKeys.COMPACT_VIEW]: 'false',
```
Comment (OrfeasZ) [inline app/reducers/user.js]: This will be ovewritten when VU hydrates the state, so it's unsafe to use it as a default.

**Review by OrfeasZ — CHANGES_REQUESTED** [review]

File: app/containers/ServerBrowser.js [hunk 2]
```diff
@@ -318,7 +320,17 @@ const mapDispatchToProps = (dispatch) => {
         },
         cycleServerSortDirection: () => {
             dispatch({ type: ActionTypes.CYCLE_SERVER_SORT_DIRECTION })
-        }
+        },
+        toggleCompactView: () => dispatch((innerDispatch, getState) => {
+            const key = AccountStorageKeys.COMPACT_VIEW
```
Comment (OrfeasZ) [inline app/containers/ServerBrowser.js]: Semicolons please :)

**Review by OrfeasZ — CHANGES_REQUESTED** [review]

File: app/reducers/user.js (hunk 1 above)
Comment (kapraran) [reply app/reducers/user.js]: hmm, ok so i'll update expecting it not to be there at all.

File: app/containers/ServerBrowser.js (hunk 2 above)
Comment (kapraran) [reply app/containers/ServerBrowser.js]: Done

File: app/reducers/user.js (hunk 1 above)
Comment (kapraran) [reply app/reducers/user.js]: Removed default state. No other changes are needed to handle this key missing.

**Pull request merged**